shop.cpp

```diff
@@ -1,26 +1,77 @@
 #include "shop.h"
-
+#include <chrono>
 
 Shop::Shop(int mMaxChairs)
     :mMaxChairs(mMaxChairs)
-    ,mTheBarber(this)
-    ,mBarberSemaphore(0)
+    ,mBarberSemaphore(1)
     ,mCustomersSemaphore(0)
 {
+    mTheBarber = new Barber(this);
```

Comment:
I think you are overusing pointers and new. If the barber is owned by the shop and lives as long as the shop lives, why not have him as a member instead of keeping a pointer?

Reply:
:ok: